### ISSUE TITLE
Make WebDriver reporter more resilient against DOM-nuking tests when writeHtml is true

### ISSUE THREAD

**Issue by kitsonk** [theintern/intern#435]
I am getting the following error when running certain tests under runner:

```
Reporter error!
TypeError: Unable to get property 'parentNode' of undefined or null reference
at suiteEnd  <__intern/lib/reporters/WebDriver.js:51:5>
at Anonymous function  <__intern/lib/ReporterManager.js:229:7>
at emit  <__intern/lib/ReporterManager.js:211:4>
at report  <__intern/lib/Suite.js:199:6>
at end  <__intern/lib/Suite.js:193:5>
at Anonymous function  <__intern/node_modules/dojo/Promise.js:156:33>
at runCallbacks  <__intern/node_modules/dojo/Promise.js:19:13>
at Anonymous function  <__intern/node_modules/dojo/Promise.js:103:21>
at run  <__intern/node_modules/dojo/Promise.js:51:33>reset
``` I suspect this is because I am modifying the DOM myself and the WebDriver reporting is getting a bit lost and is accessing the elements in an "unsafe" way.

In particular, wouldn't [this line](https://github.com/theintern/intern/blob/master/lib/reporters/WebDriver.js#L51):

```js
this.suiteNode = this.suiteNode.parentNode.parentNode || document.body;
``` Be more safely written as:

```js
this.suiteNode = this.suiteNode && this.suiteNode.parentNode && this.suiteNode.parentNode.parentNode || document.body;
```

**Comment by kitsonk** [comment]
I tried the above in a personal branch and it resolves the issue where I am running tests the cause the reference to `this.suiteNode` to be lost.

**Comment by csnover** [comment]
I presume you are wiping everything out of `document.body` to cause this? The “safer” access prevents a crash but it doesn’t do much else good, it just causes the reporter to ignore the fact that `this.suiteNode` lost its parent and retargets it to a wrong destination (`document.body`).

The more correct solution for your case is probably to set `runnerClientReporter.writeHtml` to `false` in your config (in master only at the moment). A more correct general solution would probably be to write everything starting rooted at an anonymous container node instead of `document.body`, and then if someone blows up `document.body` it can be restored because a reference will still be held.

**Comment by kitsonk** [comment]
> I presume you are wiping everything out of document.body to cause this?

Correct.

> The more correct solution for your case is probably to set runnerClientReporter.writeHtml to false in your config (in master only at the moment).

Yes, fair enough.

> A more correct patch for the reporter would probably be to write everything starting rooted at an anonymous container node instead of document.body, and then if someone blows up document.body it can be restored because a reference to the container will still be held and it can just be dumped back into the body.

Ok, let me work on that.

**Comment by devpaul** [comment]
I just experienced this issue while testing [A-Frame](https://aframe.io) since A-Frame replaces `document.body` there was no `parentNode` during `SuiteEnd`.

**Comment by kitsonk** [comment]
I will rebase #451 again! 😆

**Comment by kitsonk** [comment]
Fixed via #451 being merged
